Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -47,40 +47,68 @@
 FOUNDATION_EXPORT NSString* const NSUserDefaultsDidChangeNotification = @"NSUserDefaultsDidChangeNotification";
 
 @implementation NSUserDefaults {
+    StrongId<NSMutableDictionary> _cacheDict;
+    std::mutex _cacheLock;
+    BOOL _cacheIsDirty;
     StrongId<NSMutableDictionary> _registrationDict;
+    StrongId<NSOperationQueue> _synchronizeQueue;
 }
 
+static NSUserDefaults* _standard;
+
 /**
  @Status Interoperable
 */
 - (instancetype)init {
-    [super init];
+    return [self initWithSuiteName:nil];
+}
+
+/**
+ @Status Caveat
+ @Notes supports nil only for suitename
+*/
+- (instancetype)initWithSuiteName:(NSString*)suitename {
+    if (nil != suitename) {
+        UNIMPLEMENTED();
+        return StubReturn();
+    }
+
+    if(self = [super init]) {
 
-    _registrationDict = [NSMutableDictionary dictionary];
+        _cacheDict = [NSMutableDictionary dictionary];
+        _registrationDict = [NSMutableDictionary dictionary];
 
-    [self setObject:[NSArray arrayWithObject:@"en"] forKey:@"AppleLanguages"];
-    [self setObject:@"en_US" forKey:@"AppleLocale"];
+        _synchronizeQueue.attach([NSOperationQueue new]);
+        [_synchronizeQueue setMaxConcurrentOperationCount:1];
+
+        [self setObject:[NSArray arrayWithObject:@"en"] forKey:@"AppleLanguages"];
+        [self setObject:@"en_US" forKey:@"AppleLocale"];
+    }
 
     return self;
 }
 
+- (void)dealloc {
+    [_synchronizeQueue waitUntilAllOperationsAreFinished];
+    [super dealloc];
+}
+
 /**
  @Status Interoperable
 */
 + (NSUserDefaults*)standardUserDefaults {
-    static NSUserDefaults* standard;
-
-    if (standard == nil) {
-        standard = [self new];
+    if (_standard == nil) {
```

Comment:
The reason I avoided the [NSUserDefaults standardUserDefaults] call is to prevent initialization of standardUserDefaults when we suspend, if it hasn't already been initialized. I'll try a different workaround.

---
In reply to: [70878479](https://github.com/Microsoft/WinObjC/pull/612#discussion_r70878479) [](ancestors = 70878479,70878446)